src/runtime/plugins/vuetify-ssr-client-hints.ts

```diff
@@ -0,0 +1,176 @@
+import type { IncomingHttpHeaders } from 'node:http'
+import { type ClientHints, clientHintsConfiguration } from 'virtual:vuetify-ssr-client-hints-configuration'
+import { defineNuxtPlugin } from '#imports'
+import { useNuxtApp } from '#app'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const clientHints = clientHintsConfiguration()
+  const state = useState<SSRClientHints>('ssrClientHints')
+  const event = useRequestEvent(nuxtApp)
+
+  if (!event) {
+    if (clientHints.viewportSize) {
+      // on client, we update the display and watch for changes to avoid hydration mismatch on page refresh
+      // there will be some hydration mismatch since the headers sent by the user agent may not be accurate
+      nuxtApp.hook('app:beforeMount', () => {
+        const { width, height, update } = useNuxtApp().$vuetify.display
+        update()
+        watch(() => [width.value, height.value], ([width, height]) => {
```

Comment:
is this required? it seems if we don't update the screem (in the playground) we get hydration mismatch error, I'll test it again.

Reply:
maybe we only need the `update` call